src/index.ts

```diff
@@ -26,33 +27,61 @@ export default class CAFReceiver {
 
   // Setup DRM if present in `media.customData`
   private readonly onLoad = (loadRequestData: LoadRequestData): LoadRequestData => {
-    if (loadRequestData.media.customData && loadRequestData.media.customData.drm) {
-      return this.setDRM(loadRequestData);
+    const customData = loadRequestData.media.customData as CAFMediaInfoCustomData;
+
+    if (customData) {
+      if (customData.options) {
+        this.setWithCredentials(customData.options);
+      }
+
+      if (customData.drm) {
+        this.setDRM(customData.drm);
+      }
     }
 
     return loadRequestData;
-  }
+  };
 
-  private setDRM(loadRequestData: LoadRequestData): LoadRequestData {
-    const protectionSystem = loadRequestData.media.customData.drm.protectionSystem;
-    const licenseUrl = loadRequestData.media.customData.drm.licenseUrl;
+  private setDRM({ protectionSystem, licenseUrl, headers, withCredentials }: CAFDrmConfig): void {
     this.context.getPlayerManager().setMediaPlaybackInfoHandler((_loadRequest, playbackConfig) => {
       playbackConfig.licenseUrl = licenseUrl;
-      playbackConfig.protectionSystem =  protectionSystem;
+      playbackConfig.protectionSystem = protectionSystem as ContentProtection;
 
-      if (typeof loadRequestData.media.customData.drm.headers === 'object') {
-        playbackConfig.licenseRequestHandler = requestInfo => {
-          requestInfo.headers = loadRequestData.media.customData.drm.headers;
+      if (typeof headers === 'object') {
+        playbackConfig.licenseRequestHandler = (requestInfo) => {
+          requestInfo.headers = headers;
         };
       }
 
+      if (withCredentials) {
+        playbackConfig.licenseRequestHandler = setWithCredentialsFlag;
+      }
+
       return playbackConfig;
     });
+  }
 
-    return loadRequestData;
+  private setWithCredentials(options: CAFSourceOptions): void {
+    const playerManager = this.context.getPlayerManager();
+    const playbackConfig = Object.assign(new cast.framework.PlaybackConfig(), playerManager.getPlaybackConfig());
+
+    if (options.withCredentials) {
```

Comment:
I guess it won't make sense to also add an `option.drmWithCredentials` to allow updating the `licenseRequestHandler`?

Reply:
I don't get what you mean by `updating` it? This custom data we're talking about here is only sent initially with the load request fyi.

Reply:
Isn't this injecting a custom RequestHandler which then updates every request or the certain type? Maybe _updating_ was not the right word.
But the main question was about if we want to support `withCredentials` for license requests as well.

Reply:
Nevermind, i just realized, this is done in the `setDRM` function.

It might make more sense to move it here, but your choice.